src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkRepositoryContext.java

```diff
@@ -634,13 +635,28 @@ public StructImpl download(
             /* ensureNonEmpty= */ !allowFail,
             /* checksumGiven= */ !Strings.isNullOrEmpty(sha256)
                 || !Strings.isNullOrEmpty(integrity));
-    Optional<Checksum> checksum;
-    RepositoryFunctionException checksumValidation = null;
-    try {
-      checksum = validateChecksum(sha256, integrity, urls);
-    } catch (RepositoryFunctionException e) {
-      checksum = Optional.<Checksum>absent();
-      checksumValidation = e;
+    Checksum checksum;
+    
+    if(!sha256.isEmpty() && !integrity.isEmpty()) {
+      /* Terminates the program if both sha256 and integrity are present */
+      throw Starlark.errorf("Expected either 'sha256' or 'integrity', but not both");
+    }
+    else if(!sha256.isEmpty()) {
```

Comment:
nit: `} else ...` on one line, here and below. And put a space after `if` and before `(`